src/SmartComponents/Recs/helpers.js

```diff
@@ -76,8 +80,12 @@ export const bulkHostActions = async ({
 const getSystemCheckEndpoints = ({ ruleId, pathway }) => {
   if (ruleId) {
     return {
-      conventionalURL: `/api/insights/v1/rule/${ruleId}/systems_detail/?filter[system_profile][host_type][nil]=true&limit=1`,
-      edgeURL: `/api/insights/v1/rule/${ruleId}/systems_detail/?filter[system_profile][host_type]=edge&limit=1`,
+      conventionalURL: `/api/insights/v1/rule/${encodeURI(
+        ruleId
+      )}/systems_detail/?filter[system_profile][host_type][nil]=true&limit=1`,
+      edgeURL: `/api/insights/v1/rule/${encodeURI(
+        ruleId
+      )}/systems_detail/?filter[system_profile][host_type]=edge&limit=1`,
     };
   }
 
```

Comment:
@mhuth  "[" is also an unsafe character per RFC .https://www.ietf.org/rfc/rfc1738.txt.   Can we encodeURI the whole URL (segment) to catch all of these?

Reply:
cc @Hyperkid123  - they may be a way to handle this globally in UI framework.

Reply:
@lphiri Yes we can do this globaly. We just have to experiment a little bit to see what happens if we try to encode already encoded URI. Any changes to the global patch have obviously global impact and we have to be very careful.